src/wallet/spend.h

```diff
@@ -6,6 +6,7 @@
 #define BITCOIN_WALLET_SPEND_H
 
 #include <consensus/amount.h>
+#include <interfaces/wallet.h>
```

Comment:
This introduces a circular dependency,

`interfaces/wallet.h` is implemented by `wallet/interfaces.cpp` which includes `wallet/spend.h` which, with this change, includes `interfaces/wallet.h`.

 Thus why [suggested](https://github.com/bitcoin/bitcoin/pull/26761#discussion_r1062467148) to decouple it into a separate file.

Reply:
`wallet/spend.h` doesn't even need to include this, so it could just be removed from there and not have any issues.